src/Serilog.Extensions.Hosting/Extensions/Hosting/ReloadableLogger.cs

```diff
@@ -368,12 +368,21 @@ public bool BindProperty(string propertyName, object value, bool destructureObje
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         (ILogger, bool) UpdateForCaller(ILogger root, ILogger cached, IReloadableLogger caller, out ILogger newRoot, out ILogger newCached, out bool frozen)
         {
+            if (_frozen)
+            {
+                // If we're frozen, then the caller hasn't observed this yet and should update.
+                newRoot = _logger;
+                newCached = caller.ReloadLogger();
+                frozen = true;
+                return (newCached, true);
+            }
+
             if (cached != null && root == _logger)
             {
                 newRoot = default;
                 newCached = default;
-                frozen = _frozen;
-                return (cached, frozen); // If we're frozen, then the caller hasn't observed this yet and should update.
```

Comment:
The issue is here - even if `frozen` is true, the caching wrapper may never have seen the final logger (or any logger) - we need to propagate it as we do via line 375 in the new version.